docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +94,1028 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+ ```yaml  
+ SilverStripe\Core\Manifest\ModuleManifest:
+   project: app
+ ```
+
+To add some javascript and css files to your requirements from your PHP code, you could use this syntax:
+```php
+use SilverStripe\View\Requirements;
+
+# Load your own style and scripts
+Requirements::css('app: css/styles.css');
+Requirements::script('app: scripts/client.css');
+
+# Load some assets from a module.  
+Requirements::script('silverstripe/blog: js/main.bundle.js');
+```
+
+You can `SilverStripe\Core\Manifest\ModuleLoader` to get the web path of file.
+```php
+ModuleLoader::getModule('app')->getResource('images/road-runner.jpg')->getRelativePath();
+```
+
+You can use `SilverStripe\View\ThemeResourceLoader` to access files from your theme:
+```php
+$themeFolderPath = ThemeResourceLoader::inst()->getPath('simple');
+$themesFilePath = ThemeResourceLoader::inst()->findThemedResource('css/styles.css');
+```
+
+For classes that expect icons, you can specify theme with:
+```php
+class ListingPage extends \Page
+{
+    private static $icon = 'app: images/sitetree_icon.png';
+}
+
+class MyCustomModelAdmin extends \SilverStripe\Admin\ModelAdmin
+{
+    private static $menu_icon = 'app: images/modeladmin_icon.png';
+}
+```
+
+### Referencing static assets in template files
+SS template files accept a similar format for referencing static assets. Go through your assets files and remove hardcoded references.
+
+```html
+<img src="$ModulePath(app)/images/coyote.png" />
+<% require css("app: css/styles.css") %>
+```
+
+### Finalising removal of hardcoded paths for referencing static assets
+
+All your assets should be loading properly now.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+
+## Step 10 - Running your upgraded site for the first time {#step10}
+
+You're almost across the finish line.  
+
+### Run a dev build
+Rnn a `dev/build` either on the command line or in your browser.
+
+```bash
+/var/www/SS_acme > ./vendor/bin/sake dev/build
+```
+
+This should migrate your existing data to the new SilverStripe 4 structure.
+
+#### Migrating files
+
+Since the structure of the `File` DataObject has changed, a new task `MigrateFileTask`
+has been added to assist in migration of legacy files (see [file migration documentation](/developer_guides/files/file_migration)).
+
+```bash
+/var/www/SS_acme > ./vendor/bin/sake dev/tasks/MigrateFileTask
+```
+
+### Any other script that needs running.
+
+Some third party modules may include their own migration tasks. Take a minute to consult the release notes of your third party dependencies to make sure you have miss anything.
+
+## List of Major API Changes {#MajorAPIChange}
+This is a list of the most common API changes that might affect you.
+
+(Read the SilverStripe 4.0.0 change logs)[/changelogs/4.0.0/] for a comprehensive list of eerything that his new in SilverStripe 4.
```

Comment:
Should link to 4.0.0, 4.1.0 and 4.2.0 (only minor, not patch releases). All of those are noteworthy

Reply:
Added references to 4.0.0, 4.1.0 and 4.2.0.